### PR TITLE
Add res2net

### PR DESCRIPTION
**Pull request by ghostxsl**
- 基于`release/2.0-rc`，已对齐模型可学习参数部分的输出。
- 将anchor_generator、roi_align等运算与静态图保持一致后，精度可保持一致（在coco上Box AP为39.6，原始静态图为39.5）。
- 训练一张图像，去除随机性后，10个iteration的loss均能保持一致。

### CODE REVIEW

**Review by nemonameless — DISMISSED** [review]
LGTM

File: configs/res2net/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  timestamp = {Thu, 25 Apr 2019 10:24:54 +0200},
+  biburl    = {https://dblp.org/rec/bib/journals/corr/abs-1904-01169},
+  bibsource = {dblp computer science bibliography, https://dblp.org}
+}
+```
+
+
+## Model Zoo
+
+| Backbone                | Type           | Image/gpu | Lr schd | Inf time (fps) | Box AP | Mask AP |                           Download                           | Configs |
```
Comment (jerrywgz) [inline configs/res2net/README.md:31]: update the configs and weights

File: ppdet/modeling/backbones/res2net.py [hunk 2]
```diff
@@ -0,0 +1,330 @@
+# Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
```
Comment (jerrywgz) [inline ppdet/modeling/backbones/res2net.py]: 2020->2021

File: ppdet/modeling/backbones/res2net.py [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+@register
+@serializable
+class Res2Net(nn.Layer):
+    __shared__ = ['norm_type']
+
+    def __init__(self,
+                 depth=50,
+                 width=26,
+                 scales=4,
+                 variant='b',
```
Comment (jerrywgz) [inline ppdet/modeling/backbones/res2net.py:241]: Is variant necessary?

File: ppdet/modeling/backbones/res2net.py [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+                 groups=1,
+                 norm_type='bn',
+                 norm_decay=0,
+                 freeze_norm=True,
+                 freeze_at=0,
+                 return_idx=[0, 1, 2, 3],
+                 dcn_v2_stages=[-1],
+                 num_stages=4,
+                 in_channels=3):
+        super(Res2Net, self).__init__()
```
Comment (jerrywgz) [inline ppdet/modeling/backbones/res2net.py:251]: 需要增加注释

File: ppdet/modeling/backbones/res2net.py [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+# limitations under the License.
+
+import math
+from numbers import Integral
+
+import paddle
+import paddle.nn as nn
+import paddle.nn.functional as F
+from ppdet.core.workspace import register, serializable
+from .name_adapter import NameAdapter
```
Comment (jerrywgz) [inline ppdet/modeling/backbones/res2net.py:22]: NameAdapter 可以去除？

**Review by heavengate — DISMISSED** [review]

File: ppdet/modeling/backbones/res2net.py (hunk 4 above)
Comment (ghostxsl) [reply ppdet/modeling/backbones/res2net.py:251]: done

File: ppdet/modeling/backbones/res2net.py (hunk 3 above)
Comment (ghostxsl) [reply ppdet/modeling/backbones/res2net.py:241]: Modification completed

File: ppdet/modeling/backbones/res2net.py (hunk 2 above)
Comment (ghostxsl) [reply ppdet/modeling/backbones/res2net.py]: done

File: configs/res2net/README.md (hunk 1 above)
Comment (ghostxsl) [reply configs/res2net/README.md:31]: 我直接从静态图转过来的权重，在目前这个版本精度无法对齐，如果要对齐的话，可能需要fine-tuning或者重训

**Comment by todesti2** [comment]
请问你们的paddle版本里面的res2net有预训练模型吗
